internal/component/prometheus/mapping/mapping.go

```diff
@@ -0,0 +1,224 @@
+package mapping
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/grafana/alloy/internal/component"
+	"github.com/grafana/alloy/internal/component/prometheus"
+	"github.com/grafana/alloy/internal/featuregate"
+	"github.com/grafana/alloy/internal/service/labelstore"
+	"github.com/grafana/alloy/internal/service/livedebugging"
+	prometheus_client "github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/prometheus/model/exemplar"
+	"github.com/prometheus/prometheus/model/histogram"
+	"github.com/prometheus/prometheus/model/labels"
+	"github.com/prometheus/prometheus/model/metadata"
+	"github.com/prometheus/prometheus/storage"
+	"go.uber.org/atomic"
+)
+
+const name = "prometheus.mapping"
+
+func init() {
+	component.Register(component.Registration{
+		Name:      name,
+		Stability: featuregate.StabilityGenerallyAvailable,
+		Args:      Arguments{},
+		Exports:   Exports{},
+
+		Build: func(opts component.Options, args component.Arguments) (component.Component, error) {
+			return New(opts, args.(Arguments))
+		},
+	})
+}
+
+// Arguments holds values which are used to configure the prometheus.relabel
+// component.
+type Arguments struct {
+	// Where the relabelled metrics should be forwarded to.
+	ForwardTo []storage.Appendable `alloy:"forward_to,attr"`
+
+	// Labels to use for mapping
+	SourceLabel string `alloy:"source_label,attr"`
+
+	// Mapping
+	LabelValuesMapping map[string]map[string]string `alloy:"mapping,attr"`
```

Comment:
Is a static map the best user experience for this? Should it support reading from a file - especially using the exports of a `local.file` or `remote.s3` for example? This allows for polled updating of the static map without needing to reload/restart/directly edit the Alloy config.

Reply:
I can dynamically read the mapping using alloy components. Using this method, it avoid having parsing logic in mapping component :

```
...
local.file "mapping" {
   filename = file.path_join(module_path, "mapping.bifrost")
}

prometheus.mapping "tenants_mapping" {
  forward_to   = [prometheus.remote_write.cluster.receiver]
  source_label = "uid_label_key1"
  mapping      = encoding.from_json(local.file.mapping.content)
}
...
```

Did i miss something ?